pkg/reconciler/pipeline/controller.go

```diff
@@ -753,7 +770,14 @@ func (r *pipelineReconciler) pausePipeline(ctx context.Context, pl *dfv1.Pipelin
 	if err != nil {
 		return true, err
 	}
-	if drainCompleted {
+
+	pauseTimestamp, err := time.Parse(time.RFC3339, pl.GetAnnotations()["pauseTimestamp"])
+	if err != nil {
+		return true, err
```

Comment:
If error happens during time parsing, do not requeue  the request as it won't help.